scripts/anvil_tools/batch_upsert_entities_standard.py

```diff
@@ -72,66 +96,58 @@ def create_upsert_request(tsv, array_attr_cols=None):
+        single_entity_operations = ''''''
         # get the entity_id (row id - must be unique)
         entity_id = str(row[0])
 
-        # if array columns/attributes
+        # if array columns/attributes provided
         if array_attr_cols:
             # for each array column
             for col in array_attr_cols:
-                # add column name to template for a single attribute
-                single_attribute_request += template_make_list_attr.replace("VAR_ATTRIBUTE_LIST_NAME", col)
+                # get operation json to make the array/list attribute with column name
+                single_entity_operations += create_list_attr_operation(col)
                 # convert string value -> back into an array: [foo,bar] (str) --> ['foo', 'bar'] (list)
                 attr_values = convert_string_to_list(row[col])
-                # for each item in array, add row attribute value to the above created list attribute
+                # for each item in array (values pertaining to the list attribute as defined above)
                 for val in attr_values:
-                    single_attribute_request += template_add_list_member.replace("VAR_LIST_MEMBER", val).replace("VAR_ATTRIBUTE_LIST_NAME", col)
+                    # get operation json to add each array/list attribute's values to the request
+                    single_entity_operations += add_list_member_operation(col, val)
 
-            # get single attribute list based on which of the full tsv columns are array attributes
+            # get non-array/list attributes based on which of the full tsv columns are array attributes
             single_attr_cols = list(set(list(tsv.columns)) - set(array_attr_cols))
-        # if no array columns/attributes
+        # if no array columns/attributes provided
         else:
             single_attr_cols = list(tsv.columns)
 
-        # if there are single attribute columns - cases where all columns are arrays, single_attr_cols would be empty
+        # if there are non-array/list attribute columns - cases where all columns are arrays, single_attr_cols would be empty
         if single_attr_cols:
             # for each column that is not an array
             for col in single_attr_cols:
                 # get value in col from df
                 attr_value = str(row[col])
-                # add the request for attribute to list
-                single_attribute_request += template_make_single_attr.replace("VAR_ATTRIBUTE_MEMBER", attr_value).replace("VAR_ATTRIBUTE_NAME", col)
+                # get operation json with the row (entity) value to the non-array column (attribute)
+                single_entity_operations += create_non_array_attr_operation(col, attr_value)
 
         # remove trailing comma from the last request template
-        single_attribute_request = single_attribute_request[:-1]
+        single_entity_operations = single_entity_operations[:-1]
```

Comment:
per other comment - remove comma from the functions so you don't have to do this